### PR TITLE
MARKET-1586 ReCaptcha

### PR DESCRIPTION
**Pull request by StasivPavlo**

### CODE REVIEW

File: components/bl-reCaptcha-component/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+## Properties
+
+| Property | Type                                             | Default Value  | Logic       | Data Binding | UI Setting | Description                                                     |
+|----------|--------------------------------------------------|----------------|-------------|--------------|------------|-----------------------------------------------------------------|
+| Type     | Select ["image", "audio"]                        | "image"        |             | NO           | YES        | Allows to determine the type of verification by image or audio. |
+| Size     | Select ["compact", "normal", "invisible"]        | "normal"       | Size Logic  | YES          | YES        | Allows to determine the reCaptcha size.                         |
+| Badge    | Select ["bottom-right", "bottom-left", "inline"] | "bottom-right" |             | NO           | YES        | Allows to determine the position. Work when Size is inline.     |
+| Theme    | Select ["light", "dark"]                         | "light"        | Theme Color | YES          | YES        | Allows to determine the theme.                                  |
+
+## Action
```
Comment (addKiwi) [inline components/bl-reCaptcha-component/README.md:16]: Actions

File: components/bl-reCaptcha-component/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+| Action        | Inputs | Return                                         |
+|---------------|--------|------------------------------------------------|
+| Get Token     |        | `String`: can get when the user passed captcha |
+| Get Is Passed |        | `Boolean`: `true` when the user passed captcha |
+
```
Comment (addKiwi) [inline components/bl-reCaptcha-component/README.md:22]: I think Settings should be here
[example](https://github.com/Backendless/ui-builder-library/tree/main/components/bl-mapbox-component#settings)

File: components/bl-reCaptcha-component/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# ReCaptcha
+
+ReCaptcha is a component of Backendless UI-Builder designer. This allows you to upgrade your security from bots when need to send some form
```
Comment (v-excelsior) [inline components/bl-reCaptcha-component/README.md:3]: Please, adapt text from their site

> reCAPTCHA is a free service that protects your website from spam and abuse. reCAPTCHA uses an advanced risk analysis engine and adaptive CAPTCHAs to keep automated software from engaging in abusive activities on your site. It does this while letting your valid users pass through with ease.

File: components/bl-reCaptcha-component/README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+| Type     | Select ["image", "audio"]                        | "image"        |             | NO           | YES        | Allows to determine the type of verification by image or audio. |
+| Size     | Select ["compact", "normal", "invisible"]        | "normal"       | Size Logic  | YES          | YES        | Allows to determine the reCaptcha size.                         |
+| Badge    | Select ["bottom-right", "bottom-left", "inline"] | "bottom-right" |             | NO           | YES        | Allows to determine the position. Work when Size is inline.     |
+| Theme    | Select ["light", "dark"]                         | "light"        | Theme Color | YES          | YES        | Allows to determine the theme.                                  |
+
+## Actions
+
+| Action        | Inputs | Return                                         |
+|---------------|--------|------------------------------------------------|
+| Get Token     |        | `String`: can get when the user passed captcha |
```
Comment (v-excelsior) [inline components/bl-reCaptcha-component/README.md:20]: On Token Sent
On Pass(ed) - not sure which variant is correct

File: components/bl-reCaptcha-component/README.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+```
+{
+    "success": true|false,
+    "challenge_ts": timestamp,  // timestamp of the challenge load (ISO format yyyy-MM-dd'T'HH:mm:ssZZ)
+    "hostname": string,         // the hostname of the site where the reCAPTCHA was solved
+    "error-codes": [...]        // optional
+}
+```
+![](example-images/create-method-example.jpg)
+![](example-images/api-service-captcha-codeless.jpg)
```
Comment (v-excelsior) [inline components/bl-reCaptcha-component/README.md:57]: Hide your test key from this image

File: components/bl-reCaptcha-component/src/index.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  compact  : CaptchaComponent,
+  normal   : CaptchaComponent,
+  invisible: InvisibleCaptcha
+};
+
+export default function Captcha({ component, settings }) {
+  const { display, style, classList, size } = component;
+  const { siteKey } = settings;
+
+  const Captcha = SizeCaptcha[size];
```
Comment (v-excelsior) [inline components/bl-reCaptcha-component/src/index.js:16]: size -> type

File: components/bl-reCaptcha-component/src/index.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div className={ cn('bl-customComponent-reCaptcha', classList) } style={ style }>
```
Comment (v-excelsior) [inline components/bl-reCaptcha-component/src/index.js:23]: If it not in popover when visible - add root ref(component.el)

File: components/bl-reCaptcha-component/src/index.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
+import { useRef, useState, useCallback } from 'react';
+import { ReCaptcha } from './re-captcha';
```
Comment (v-excelsior) [inline components/bl-reCaptcha-component/src/index.js:2]: Need to add an error callback with action On Error

**Review by v-excelsior — CHANGES_REQUESTED** [review]

**Review by v-excelsior — CHANGES_REQUESTED** [review]

**Pull request merged**